src/daft-core/src/python/series.rs

```diff
@@ -35,35 +35,39 @@ pub struct PySeries {
 #[pymethods]
 impl PySeries {
     #[staticmethod]
-    pub fn from_arrow(py: Python, name: &str, pyarrow_array: Bound<PyAny>) -> PyResult<Self> {
+    #[pyo3(signature = (name, pyarrow_array, dtype=None))]
+    pub fn from_arrow(
+        py: Python,
+        name: &str,
+        pyarrow_array: Bound<PyAny>,
+        dtype: Option<PyDataType>,
+    ) -> PyResult<Self> {
         let arrow_array = ffi::array_to_rust(py, pyarrow_array)?;
         let arrow_array = cast_array_for_daft_if_needed(arrow_array.to_boxed());
-        let series = series::Series::try_from((name, arrow_array))?;
+
+        let series = if let Some(dtype) = dtype {
+            let field = Field::new(name, dtype.into());
+            series::Series::try_from_field_and_arrow_array(field, arrow_array)?
+        } else {
+            series::Series::try_from((name, arrow_array))?
+        };
+
         Ok(series.into())
     }
 
     // This ingests a Python list[object] directly into a Rust PythonArray.
     #[staticmethod]
-    pub fn from_pylist(
-        py: Python<'_>,
-        name: &str,
-        pylist: Bound<PyAny>,
-        pyobj: &str,
-    ) -> PyResult<Self> {
+    pub fn from_pylist(name: &str, pylist: Bound<PyAny>, dtype: PyDataType) -> PyResult<Self> {
         let vec_pyobj: Vec<PyObject> = pylist.extract()?;
-        let dtype = match pyobj {
-            "force" => DataType::Python,
-            "allow" => infer_daft_dtype_for_sequence(&vec_pyobj, py, name)?.unwrap_or(DataType::Python),
-            "disallow" => panic!("Cannot create a Series from a pylist and being strict about only using Arrow types by setting pyobj=disallow"),
-            _ => panic!("Unsupported pyobj behavior when creating Series from pylist: {}", pyobj)
-        };
+
         let vec_pyobj_arced = vec_pyobj.into_iter().map(Arc::new).collect();
         let arrow_array: Box<dyn arrow2::array::Array> =
             Box::new(PseudoArrowArray::from_pyobj_vec(vec_pyobj_arced));
-        let field = Field::new(name, DataType::Python);
 
+        let field = Field::new(name, DataType::Python);
         let data_array = DataArray::<PythonType>::new(field.into(), arrow_array)?;
-        let series = data_array.cast(&dtype)?;
+        let series = data_array.cast(&dtype.into())?;
```

Comment:
I feel like we could take this even further to avoid jumping between Rust and Python by doing as much casting upfront as possible, but it was starting to get too complicated so fine for now